pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -125,8 +115,8 @@ func NewFluxAddonClient(flux Flux, gitOpts *GitOptions) *FluxAddonClient {
 	}
 	return &FluxAddonClient{
 		flux:              flux,
-		gitProviderClient: gitOpts.Git,
-		gitClient:         gitOpts.Git,
+		gitProviderClient: gitOpts.GitProvider,
```

Comment:
should we rename this to `gitProvider` too?

Reply:
what do u think abt having just one `git` as `FluxAddonClient` field?

```go
type git struct {
  provider
  client
  writer
}
```

Reply:
yeah I like that I think it would make it much more clear

Reply:
this is now `GitTools`; it's returned from the factory, input to the `FluxAddonClient`, and used throughout.